### ISSUE TITLE
Filtering of stored procedures

### ISSUE THREAD

**Issue by ErikEJ** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#83]
Hi, is there an easy way to filter stored procedures like this: include all stored procedures in a schema abc and also some tables from the dbo schema?

**Comment by sjh37** [comment]
Hi @ErikEJ I have made a small change to the SP filters (pull in the latest code) to make them work the same way as it does for tables. I should of done this previously.

``` c#
string spName = rdr["SPECIFIC_NAME"].ToString().Trim();
var fullname = schema + "." + spName;
if(storedProcedureFilterExclude != null && (
   storedProcedureFilterExclude.IsMatch(spName) || 
   storedProcedureFilterExclude.IsMatch(fullname)))
    continue;
```

So it now checks for the **name** of the stored procedure, and **schema.name**. 
So you could exclude non abc schemas at this point by having a regex to find the start of the string as

And it now also checks after the renaming of the stored procedure:

``` c#
sp.NameHumanCase = StoredProcedureRename(sp.NameHumanCase, schema);
if(storedProcedureFilterExclude != null && 
   storedProcedureFilterExclude.IsMatch(sp.NameHumanCase))
    continue;
```

So you could use the rename callback function to add XXX to the start of the proc name, and have XXX excluded in the regex filter.

**Comment by ErikEJ** [comment]
You are a star 👏

**Comment by sjh37** [comment]
Released in v2.23.0
